internal/app/query/query_service.go

```diff
@@ -135,7 +135,53 @@ func (s *queryService) QueryTicketIds(req *pb.QueryTicketIdsRequest, responseSer
 }
 
 func (s *queryService) QueryBackfills(req *pb.QueryBackfillsRequest, responseServer pb.QueryService_QueryBackfillsServer) error {
-	return status.Error(codes.Unimplemented, "not implemented")
+	if req == nil {
+		return status.Error(codes.InvalidArgument, "request entry cannot be nil")
+	}
+
+	pool := req.GetPool()
+	if pool == nil {
+		return status.Error(codes.InvalidArgument, ".pool is required")
+	}
+
+	pf, err := filter.NewPoolFilter(pool)
+	if err != nil {
+		return err
+	}
+
+	ctx := responseServer.Context()
+	var results []*pb.Backfill
+	err = s.bfc.requestBackfills(ctx, func(backfills map[string]*pb.Backfill) {
+		for _, backfill := range backfills {
+			if pf.In(backfill) {
+				results = append(results, backfill)
+			}
+		}
+	})
+	if err != nil {
+		err = errors.Wrap(err, "QueryBackfills: failed to run request")
+		return err
```

Comment:
@hsorellana I mean should you return here some status codes?

Reply:
ouh you're right!